File: g3malt/src/main/java/isaapp/g3malt/dto/UserCredentialsDTO.java
package isaapp.g3malt.dto;

import javax.validation.constraints.*;

public class UserCredentialsDTO {
	@NotNull
	@NotEmpty
	@Email
	public String email;
	@NotNull
	@NotEmpty
	@Size(min=5)
	public String password;
	public int userId;
	public UserCredentialsDTO(String email, String password,int userId) {
		this.email = email;
		this.password = password;
		this.userId = userId;
	}

	public UserCredentialsDTO() {}

	public String getEmail() {
		return email;
	}

	public void setEmail(String email) {
		this.email = email;
	}

	public String getPassword() {
		return password;
	}

	public void setPassword(String password) {
		this.password = password;
	}

	public int getUserId() {
		return userId;
	}

	public void setUserId(int userId) {
		this.userId = userId;
	}
}

File: frontend/src/app/DTO/user-credentials-dto.ts
export class UserCredentialsDTO {
    [x: string]: any;
    email: String = "";
    password: String = "";
    userId: number = 0;

    public constructor(obj?: any) {
        if (obj) {
            this.email = obj.email;
            this.password = obj.password;
            this.userId = obj.userId;
        }
    }
}
